crates/pessimistic-proof-program/src/main.rs

```diff
@@ -1,16 +1,23 @@
 #![no_main]
 
+use bincode::Options;
 use pessimistic_proof::local_exit_tree::hasher::Keccak256Hasher;
 use pessimistic_proof::multi_batch_header::MultiBatchHeader;
 use pessimistic_proof::{generate_pessimistic_proof, LocalNetworkState};
 
 sp1_zkvm::entrypoint!(main);
-
 pub fn main() {
     let initial_state = sp1_zkvm::io::read::<LocalNetworkState>();
     let batch_header = sp1_zkvm::io::read::<MultiBatchHeader<Keccak256Hasher>>();
 
     let outputs = generate_pessimistic_proof(initial_state, &batch_header).unwrap();
 
-    sp1_zkvm::io::commit(&outputs);
+    let pp_inputs = bincode::DefaultOptions::new()
+        .with_big_endian()
+        .with_fixint_encoding()
+        .allow_trailing_bytes()
```

Comment:
Why do we need this one?

Reply:
The motivation is to avoid trimmed prefixed zeroes, but that's already guaranteed by the `with_fixint_encoding` so I removed the `allow_trailing_bytes`